### PR TITLE
Added ShanghaiTech University Faculty

### PR DESCRIPTION
**Pull request by Dzhange**
# Contributing to CSrankings

Thanks for contributing to CSrankings! Here are some guidelines to getting your pull request accepted.

0. Use a reasonable title that explains what the PR corresponds to (as in, not "Update csrankings-x.csv"). **IF YOU DO NOT DO THIS, YOUR COMMIT WILL BE SUMMARILY REJECTED.**

1. _Do not use Excel to edit any .csv files; Excel incorrectly tries to
convert some Google Scholar entries to formulas, corrupting the
database. Use a text editor like emacs or NotePad instead._

2. _Insert new faculty **in alphabetical order**, not at the end of `csrankings-[a-z].csv`._ **Do not modify `csrankings.csv`, which is auto-generated.** **DO NOT MODIFY `csrankings-[0-9].csv`, WHICH ARE BEING DEPRECATED. USE `csrankings-[a-z].csv` INSTEAD.**

3. _Read and check **all** the boxes below by filling them in with an X._

**Inclusion criteria**

- [x] Make sure that any faculty you add meet the inclusion
criteria. Eligible faculty include only full-time, tenure-track research
faculty members on a given campus who can *solely* advise PhD students in
Computer Science. Faculty not in a CS department or similar who can
advise PhD students in CS can be included regardless of their home
department. Faculty should also have a 75%+ time appointment (check
`old/industry.csv` for faculty who are now more than 25% in industry).

**Updating an affiliation or home page**

- [x] Update affiliations, home pages, and Google Scholar entries by modifying `csrankings-[a-z].csv`. For the Google Scholar entry, just use the alphanumeric identifier in the middle of the URL. If none is there, put NOSCHOLARPAGE.

**Adding one or more faculty members (including an entire department)**

- [x] If the department is not yet listed in CSrankings, the entire faculty needs to be added (not just one faculty member).

- [x] Enter each faculty member's [DBLP](http://dblp.org) name, home page, and Google Scholar entry (just the alphanumeric identifier, not the whole URL) by modifying `csrankings-[a-z].csv` (**the letters correspond to the first letter of the faculty members' names**; please **do not** use the numeric suffixes, which are being obsoleted); include disambiguation suffixes like 0001 as needed. If the faculty entry is currently ambiguous, please do not include them. Send mail to the DBLP maintainers (dblp@dagstuhl.de) with a few publications by a particular faculty member; also, open an issue so that when the DBLP database is updated, that faculty member's information can be added.

- [x] If DBLP has multiple entries for this person, all of them need to be listed. If an alias is not already present in `dblp-aliases.csv`, add it.

- [x] If the institution you are adding is not in the US,
update `country-info.csv`.

**(Advanced) Quick contribution via a shallow clone** 

A full clone of the CSrankings repository is almost 2GB.  To contribute a change without creating a full local clone of the CSrankings repo, you can do a _shallow clone_.  To do so, follow these steps:

1. Fork the CSrankings repo.  If you have an existing fork, but it is not up to date with the main repository, this technique may not work.  If necessary, delete and re-create your fork to get it up to date.  (Do **not** delete your existing fork if it has unmerged changes you want to preserve!)
2. Do a shallow clone of your fork: `git clone --depth 1 https://github.com/yourusername/CSrankings`.  This will only download the most recent commit, not the full git history.
3. Make your changes on a branch, push them to your clone, and create a pull request on GitHub as usual.

If you want to do another contribution and some time has passed, perform steps 1-3 again, creating a fresh fork and shallow clone.

### CODE REVIEW

File: csrankings-f.csv [hunk 1]
```diff
@@ -567,6 +567,7 @@ Fritz Henglein,University of Copenhagen,http://www.diku.dk/~henglein,FOM81UEAAAA
 Fritz Kevin Flores,De La Salle University,https://www.dlsu.edu.ph/colleges/ccs/faculty-profile/single/?id=32665778068&command=GETPROFILE,NOSCHOLARPAGE
 Froduald Kabanza,Université de Sherbrooke,https://www.usherbrooke.ca/informatique/personnel/corps-professoral/professeurs/froduald-kabanza,h7z0lEoAAAAJ
 Frouke Hermens,Tilburg University,https://www.tilburguniversity.edu/webwijs/show/f.hermens.htm,oTRqgI4AAAAJ
+Fu Song, ShanghaiTech University,https://sist.shanghaitech.edu.cn/sist_en/2020/0814/c7582a54913/page.htm,RTTSafIAAAAJ
```
Comment (emeryberger) [inline csrankings-f.csv]: Remove spurious space after comma.

File: csrankings-j.csv [hunk 2]
```diff
@@ -1146,6 +1147,7 @@ Jingtao Yao,University of Regina,http://www.cs.uregina.ca/People/Profiles/JingTa
 Jinguang Han,Queen's University Belfast,https://www.qub.ac.uk/schools/eeecs/Connect/Staff/BusinessCard/?name=j.han,euLvHOkAAAAJ
 Jingwei Li,UESTC,https://jingwei87.github.io,uSv67rIAAAAJ
 Jingwen Leng,Shanghai Jiao Tong University,http://www.jingwenleng.com/home/me.html,L1y8y2MAAAAJ
+Jingya Wang, ShanghaiTech University,https://sist.shanghaitech.edu.cn/2020/0707/c7499a58578/page.htm,vmvJV_IAAAAJ
```
Comment (emeryberger) [inline csrankings-j.csv]: Remove spurious space after comma.

File: csrankings-s.csv [hunk 3]
```diff
@@ -818,6 +818,7 @@ Sheng-Fu Liang,National Cheng Kung University,http://www.csie.ncku.edu.tw/ncku_c
 Sheng-Hua Zhong,Shenzhen University,https://sheng-hua-zhong.weebly.com,O6XQ9c4AAAAJ
 Sheng-Tzong Cheng,National Cheng Kung University,http://www.csie.ncku.edu.tw/ncku_csie/depmember/teacherdetail/id/9,NOSCHOLARPAGE
 Sheng-hua Zhong,Shenzhen University,https://sheng-hua-zhong.weebly.com,O6XQ9c4AAAAJ
+Shenghua Gao,ShanghaiTech University, https://svip-lab.github.io/news.html,fe-1v0MAAAAJ
```
Comment (emeryberger) [inline csrankings-s.csv]: Remove spurious space after comma.

**Review by emeryberger — CHANGES_REQUESTED** [review]

**Comment by Dzhange** [comment]
Removed spurious space and resolved conflicts, please check, thanks

**Comment by emeryberger** [comment]
Please re-pull a, c, and d, which are all not conflicting but every single line is different (presumably because of line ending issues).

**Comment by Dzhange** [comment]
Thanks for the instruction. I just pulled, does it work now?

**Comment by emeryberger** [comment]
I still see a, c, and d with huge diffs.

**Comment by Dzhange** [comment]
sorry let me try it again

**Comment by Dzhange** [comment]
I just downloaded the a,c,d files directly from the original repo and committed again, but there still is huge diff, kind of weird

**Comment by Dzhange** [comment]
And when I try to pull from the upstream it says "Already up to date"

**Comment by emeryberger** [comment]
how about you just remove the a,c,d files from this PR so I can merge it, and you can open another PR?

**Comment by Dzhange** [comment]
Yeah let me try that

**Comment by emeryberger** [comment]
Unfortunately, this merge would *delete* the above-mentioned files :(

**Comment by Dzhange** [comment]
Oh no, does that mean it would delete those files in the origin repo?

**Comment by Dzhange** [comment]
Another solution is I just close this pull request and redo all the editing, should take less time than figuring out this mystery

**Comment by emeryberger** [comment]
Yeah, agreed, let's close it and start over

**Pull request closed without merging**